scipy/sparse/tests/test_base.py

```diff
@@ -5156,11 +5277,24 @@ def test_setdiag_noncanonical(self):
         assert_(np.all(np.diff(m.col) >= 0))
 
 
-def cases_64bit():
-    TEST_CLASSES = [TestBSR, TestCOO, TestCSC, TestCSR, TestDIA,
-                    # lil/dok->other conversion operations have get_index_dtype
-                    TestDOK, TestLIL
-                    ]
+class TestCOONonCanonicalMatrix(TestCOONonCanonical, TestCOOMatrix):
+    pass
+
+
+def cases_64bit(sp_api):
+    if sp_api == "sparray":
+        TEST_CLASSES = [TestBSR, TestCOO, TestCSC, TestCSR, TestDIA,
+                         # lil/dok->other conversion operations have get_index_dtype
```

Comment:
What does this comment mean?

Reply:
It was there before -- and I interpret it to mean:
LIL and DOK will not use get_index_dtype directly, but they do sometimes convert to the CSR or other formats which **do** use get_index_dtype. So even though it might seem wasteful, they include the TestLIL and TestDOK tests in this collection process just in case they are impacted by get_index_dtype.

My checks indicate that there are 721 tests run as part of each method using `cases_64bit` in Test64Bit that do not even call `get_index_dtype`. There is no reason to call those methods except ease of gathering the tests. And listing them all somewhere is a possibility but I am looking for a better (and more maintainable) way to reduce test duplication here than listing methods that don't use it.

I suspect these tests are mostly overkill in response to angst about 32-bit vs 64-bit machines. That is technical debt that I have tried to find a way to eliminate. We should really only be testing cases where we suspect something **could** go wrong. But that's a hard thing to identify -- and removing tests requires a high bar of assurance (reasonably so).

Reply:
Thanks for explaining! Since this is not part of your PR, it's not expected for you to clarify, but it would be helpful to have some text that outlines this reasoning; otherwise, that comment is mostly meaningless.

Reply:
I expanded the comments explaining cases_64bit:
```
+#Todo: Revisit 64bit tests: avoid rerun of all tests for each version of get_index_dtype
 def cases_64bit(sp_api):
+    """Yield all tests for all formats that use get_index_dtype
+
+    This is more than testing get_index_dtype. It allows checking whether upcasting
+    or downcasting the index dtypes affects test results. The approach used here
+    does not try to figure out which tests might fail due to 32/64-bit issues.
+    We just run them all.
+    So, each test method in that uses cases_64bit reruns most of the test suite!
+    """
     if sp_api == "sparray":
         TEST_CLASSES = [TestBSR, TestCOO, TestCSC, TestCSR, TestDIA,
-                         # lil/dok->other conversion operations have get_index_dtype
+                         # lil/dok->other conversion operations use get_index_dtype
+                         # so we include lil & dok test suite even though they do not
+                         # use get_index_dtype within the class. That means many of
+                         # these tests are superfluous, but it's hard to pick which
```

Reply:
Thanks, Dan!